src/serve/handlers.rs

```diff
@@ -0,0 +1,306 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! HTTP request handlers for the serve command.
+
+use std::sync::Arc;
+
+use axum::{
+    extract::{Path, Query, State},
+    http::StatusCode,
+    response::IntoResponse,
+    Json,
+};
+use serde_json::json;
+
+use super::server::AppState;
+use super::types::{RegistryCounts, RegistryOverview, SearchParams, SearchResponse};
+
+/// Health check endpoint.
+#[utoipa::path(
+    get,
+    path = "/health",
+    responses(
+        (status = 200, description = "Service is healthy")
+    ),
+    tag = "health"
+)]
+pub async fn health() -> StatusCode {
+    StatusCode::OK
+}
+
+/// Serve a specific schema by name.
+#[utoipa::path(
+    get,
+    path = "/api/v1/schema/{name}",
+    params(
+        ("name" = String, Path, description = "Schema name (forge or semconv)")
+    ),
+    responses(
+        (status = 200, description = "Requested schema", content_type = "application/json"),
+        (status = 404, description = "Schema not found")
+    ),
+    tag = "schemas"
+)]
+pub async fn get_schema(Path(name): Path<String>) -> impl IntoResponse {
+    let name = name.trim_start_matches('/');
+
+    let schema = match name {
+        "forge" => include_str!("../../schemas/forge.schema.v2.json"),
```

Comment:
Nit: Take a look at `weaver registry schema` command, we can actually generate these strings from Schemars, and I'd suggest we look at doing that.

Reply:
Changed to generate schemas on the fly